### ISSUE TITLE
use smolVLA, How to know the current task is completed

### ISSUE THREAD

**Issue by haoyankai** [huggingface/lerobot#1262]
I use smolVLA to do a wiping task, it will keep doing the task again and again, how to judge the task is completed, thank you

**Comment by n3puiol** [comment]
SmolVLA uses SmolVLM. You could try using it either separately (as shown here https://huggingface.co/HuggingFaceTB/SmolVLM2-500M-Video-Instruct) but you should probably just reuse the one from SmolVLA:
```py
policy = SmolVLAPolicy.from_pretrained("lerobot/smolvla_base")
model = policy.model.vlm_with_expert
processor = model.processor

image = <image_from_your_environment>
prompt = "Is task <your_task> complete?"

inputs = processor(images=image, text=prompt, return_tensors="pt", padding=True)
image_tensor = inputs["pixel_values"]  # shape: [B, C, H, W]
input_ids = inputs["input_ids"]        # shape: [B, L]

image_embeds = model.embed_image(image_tensor)
text_embeds = model.embed_language_tokens(input_ids)

batch_size, seq_len = input_ids.shape
attention_mask = torch.ones((batch_size, seq_len + image_embeds.shape[1], seq_len + image_embeds.shape[1]), dtype=torch.bool)
position_ids = torch.arange(seq_len + image_embeds.shape[1]).unsqueeze(0).repeat(batch_size, 1)

inputs_embeds = [image_embeds, text_embeds]
outputs, _ = model.forward(
    attention_mask=attention_mask,
    position_ids=position_ids,
    inputs_embeds=inputs_embeds,
    use_cache=False,
    fill_kv_cache=False,
)

print(outputs[1])
```

### Note!!
I haven't tried this but hopefully it helps.

**Comment by haoyankai** [comment]
> SmolVLA uses SmolVLM. You could try using it either separately (as shown here https://huggingface.co/HuggingFaceTB/SmolVLM2-500M-Video-Instruct) but you should probably just reuse the one from SmolVLA:
> 
> policy = SmolVLAPolicy.from_pretrained("lerobot/smolvla_base")
> model = policy.model.vlm_with_expert
> processor = model.processor
> 
> image = <image_from_your_environment>
> prompt = "Is task <your_task> complete?"
> 
> inputs = processor(images=image, text=prompt, return_tensors="pt", padding=True)
> image_tensor = inputs["pixel_values"]  # shape: [B, C, H, W]
> input_ids = inputs["input_ids"]        # shape: [B, L]
> 
> image_embeds = model.embed_image(image_tensor)
> text_embeds = model.embed_language_tokens(input_ids)
> 
> batch_size, seq_len = input_ids.shape
> attention_mask = torch.ones((batch_size, seq_len + image_embeds.shape[1], seq_len + image_embeds.shape[1]), dtype=torch.bool)
> position_ids = torch.arange(seq_len + image_embeds.shape[1]).unsqueeze(0).repeat(batch_size, 1)
> 
> inputs_embeds = [image_embeds, text_embeds]
> outputs, _ = model.forward(
>     attention_mask=attention_mask,
>     position_ids=position_ids,
>     inputs_embeds=inputs_embeds,
>     use_cache=False,
>     fill_kv_cache=False,
> )
> 
> print(outputs[1])
> 
> ### Note!!
> 
> I haven't tried this but hopefully it helps.

Thank you for your response, I will try it

**Comment by Sergeytata** [comment]
Some datasets from Lerobot collection have done state. It’s sent at the end of the sequence. I wonder if it can be added as an option to Lerobot record script so it’s added to the state. It will solve this problem.

**Comment by haoyankai** [comment]
> Some datasets from Lerobot collection have done state. It’s sent at the end of the sequence. I wonder if it can be added as an option to Lerobot record script so it’s added to the state. It will solve this problem.

Thank you for your advice, i will try it
